v3/src/components/graph/components/histogram.tsx

```diff
@@ -0,0 +1,143 @@
+    // build and render bar cover elements that will handle click events for the fused points
+    if (dataConfig && abovePointsGroupRef?.current) {
+      const uniqueBarCovers = new Map<string, IBarCover>()
+      Object.keys(bins).forEach(primeCat => {
+        Object.keys(bins[primeCat]).forEach(secCat => {
+          Object.keys(bins[primeCat][secCat]).forEach(primeSplitCat => {
+            Object.keys(bins[primeCat][secCat][primeSplitCat]).forEach(secSplitCat => {
+              const cellData = bins[primeCat][secCat][primeSplitCat]
+              cellData.forEach((cell, cellIndex) => {
+                if (cell.length === 0) return
+                const barSecondaryDimension = secondaryNumericUnitLength * cell.length
+                const barWidth = primaryIsBottom
+                                   ? primaryScaleDiff / numExtraPrimaryBands
+                                   : barSecondaryDimension
+                const barHeight = primaryIsBottom
+                                    ? barSecondaryDimension
+                                    : -primaryScaleDiff / numExtraPrimaryBands
+                const x = primaryIsBottom
+                            ? getPrimaryScreenCoord(cell[0]) - barWidth / 2
+                            : getSecondaryScreenCoord(cell[0]) - secondaryNumericUnitLength / 2
+                const y = primaryIsBottom
+                            ? getSecondaryScreenCoord(cell[0]) - barHeight + secondaryNumericUnitLength / 2
+                            : getPrimaryScreenCoord(cell[0]) - barHeight / 2
+                const key = `${primeCat}-${secCat}-${primeSplitCat}-${cellIndex}`
+                const cover = {
+                  caseIDs: cell,
+                  class: `bar-cover bar-cover-${cellIndex} draggable-bin-boundary`,
+                  primeCat, secCat, primeSplitCat, secSplitCat,
+                  x: x.toString(), y: y.toString(),
+                  width: barWidth.toString(), height: barHeight.toString()
+                }
+                if (!uniqueBarCovers.has(key)) uniqueBarCovers.set(key, cover)
+              })
+            })
+          })
+        })
+      })
+      const barCovers: IBarCover[] = Array.from(uniqueBarCovers.entries()).map(([_, cover]) => cover)
+      renderBarCovers({ barCovers, barCoversRef, dataConfig, primaryAttrRole })
+    }
+
+    setPointCoordinates({
+      pointRadius: graphModel.getPointRadius(),
+      selectedPointRadius: graphModel.getPointRadius("select"),
+      pixiPoints, selectedOnly, pointColor, pointStrokeColor, getWidth, getHeight,
+      getScreenX, getScreenY, getLegendColor, getAnimationEnabled: isAnimating,
+      pointDisplayType, dataset, pointsFusedIntoBars: true
+    })
+  }, [abovePointsGroupRef, dataConfig, dataset, getPrimaryScreenCoord, getSecondaryScreenCoord, graphModel,
+      isAnimating, layout, pixiPoints, pointColor, pointStrokeColor, primaryAttrRole, primaryAxisScale,
+      primaryIsBottom, primaryPlace, secondaryAttrRole])
```

Comment:
Because `getPrimaryScreenCoord` and `getSecondaryScreenCoord` are unstable (as pointed out elsewhere), `refreshPointPositions` will be re-instantiated on every render as well.